### PR TITLE
CLC-5837 OEC: Reorganize validate/export/publish tasks

### PR DESCRIPTION
**Pull request by paulkerschen**
https://jira.ets.berkeley.edu/jira/browse/CLC-5837

### CODE REVIEW

File: app/models/oec/export_and_publish_task.rb [hunk 1]
```diff
@@ -1,14 +0,0 @@
-module Oec
-  class ExportAndPublishTask < Task
-
```
Comment (johncrossman) [inline app/models/oec/export_and_publish_task.rb]: Good riddance!

File: app/models/oec/merged_sheet_validation.rb [hunk 2]
```diff
@@ -1,9 +1,9 @@
...
-  class ExportTask < Task
+  module MergedSheetValidation
 
     include Validator
 
-    def run_internal
+    def build_and_validate_export_sheets
       merged_course_confirmations = @remote_drive.find_nested [@term_code, 'departments', 'Merged course confirmations']
       merged_course_confirmations_csv = merged_course_confirmations && @remote_drive.export_csv(merged_course_confirmations)
       raise RuntimeError, 'No merged course confirmation sheet found' unless merged_course_confirmations_csv
```
Comment (johncrossman) [inline app/models/oec/merged_sheet_validation.rb]: If we pass "on_failure: :error" to *find_nested(... 'Merged course confirmations')* can we do away with the above nil check / raise Error?

File: app/models/oec/publish_task.rb [hunk 3]
```diff
@@ -1,64 +1,62 @@
...
+      pattern = "#{Oec::Task.date_format}_%H%M%S"
+      csv_staging_dir = @tmp_dir.join("publish_#{@date_time.strftime pattern}")
+      FileUtils.mkdir_p csv_staging_dir
+
+      files_to_publish = []
+      export_sheets.each do |sheet|
+        sheet.export_directory = csv_staging_dir
+        sheet.write_csv
+        files_to_publish << "#{sheet.export_name}.csv"
+      end
```
Comment (johncrossman) [inline app/models/oec/publish_task.rb]: Nice! Cleaner.

**Comment by johncrossman** [comment]
I posted one question. Otherwise, ready to merge. Thanks, @paulkerschen

File: app/models/oec/merged_sheet_validation.rb (hunk 2 above)
Comment (paulkerschen) [inline app/models/oec/merged_sheet_validation.rb]: @johncrossman Yes, well caught. I'll amend.

**Comment by paulkerschen** [comment]
@johncrossman, pushed amendment to pass `on failure: :error` when pulling the two merged confirmation sheets; T chawing it.

**Comment by johncrossman** [comment]
Travis chewed then spat out success.

**Pull request merged**